rootfs/etc/nginx/lua/balancer.lua

```diff
@@ -92,15 +96,19 @@ function _M.init_worker()
 end
 
 function _M.call()
-  ngx_balancer.set_more_tries(1)
-
-  local host, port = balance()
-
-  local ok, err = ngx_balancer.set_current_peer(host, port)
-  if ok then
-    ngx.log(ngx.INFO, "current peer is set to " .. host .. ":" .. port)
-  else
-    ngx.log(ngx.ERR, "error while setting current upstream peer to: " .. tostring(err))
+  local phase = ngx.get_phase()
+  
+  if phase = "balancer" then
+    ngx_balancer.set_more_tries(1)
+    local host, port = balance()
+    local ok, err = ngx_balancer.set_current_peer(host, port)
+    if ok then
+      ngx.log(ngx.INFO, "current peer is set to " .. host .. ":" .. port)
+    else
+      ngx.log(ngx.ERR, "error while setting current upstream peer to: " .. tostring(err))
+    end
+  elseif phase == "log" then
+    ewma.after_balance()
```

Comment:
`call` function is currently being called only from `balancer_by_lua` block, i.e balancer phase only. So `ngx.get_phase()` will never be `log`. We will have to add `log_by_lua_block` to call this method.

Let's add that block to https://github.com/Shopify/ingress/blob/master/rootfs/etc/nginx/template/nginx.tmpl#L631 if `$all.DynamicConfigurationEnabled` is true and call `balancer.after_balance()` where `balancer.after_balance()` will call `ewma.after_balance()` if the `lb_alg` is `ewma`.